nb-api/src/main/java/io/nosqlbench/components/NBBaseComponent.java

```diff
@@ -122,6 +126,14 @@ public final void close() throws RuntimeException {
         }
     }
 
+    public void reportExecutionMetric(NBMetric m) {
+        if (parent != null) {
+            parent.reportExecutionMetric(m);
+        } else {
+            logger.debug("buffering metric " + m.toString() + " for final report");
```

Comment:
This log line doesn't match with any other logic I can see here.

Reply:
good point. Changed it to a warn level message about encountering a null parent outside of NBSession.